### DISCUSSION TITLE
tsinfer verify throw an error: ValueError: Alleles for sample 6 not equal at site 4

### DISCUSSION THREAD

**Discussion by chlangley** [tskit-dev/tsinfer#624] [Q&A] [answered]
Hello:
I have stumbled around this early impasse long enough.  Time for a clue, please.

While I have gotten my data into a valid Sample and been able to add metadata (sampleIDs and populationIDs) and have run tsinfer infer on both the entire chromosome and on a small sub regions, test_direct.samples (see attached [tsinfer_infer_out.txt](https://github.com/tskit-dev/tskit/files/7885121/tsinfer_infer_out.txt).  Note the unnerving line 13 in 'tsinfer_infer_out.txt' where the progress begins unexpectedly: "... sites; 7301 mutations######## ...".

When I examine indivdual trees on a simplify(d) subsample they appear to fit my expectations from previous analyses of these data.  

Yet the resulting test_direct.trees does not tsinfer verify with test_direct.samples (see [tsinfer_verify_out.txt](https://github.com/tskit-dev/tskit/files/7885127/tsinfer_verify_out.txt) .  This seems like a good reason to pause and find out the meaning of the deviant genotypes.

The input file looks like this in the upper lefthand corner (snippet) where one might expect the " Alleles for sample 6 not equal at site 4" to be.
`| => less ../cpragts/kgp4_chrX_hap_ba_snp`
`2781487 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`
`2781514 0 0 1 0 1 1 0 0 0 1 0 0 0 0 1 0 0`
`2781518 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`
`2781584 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0`
`2781603 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`
`2781604 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`
`2781623 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`
`2781635 1 0 0 0 0 0 0 0 1 0 1 0 1 0 0 1 0`
`2781642 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`
`2781670 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`
`2781698 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`
`2781865 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`
`2781878 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`
`2781893 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`
`2781927 0 1 1 1 1 0 1 0 0 0 0 1 0 1 0 0 1`
`2781983 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0`

 I don't see an obvious candidate.  Everything is 0,1,2 or -1.
How might one proceed to determine the source of this error?  

thanks for any help.
Cheers,
Chuck

**Comment by molpopgen** [comment]
I think this belongs over at `tsinfer`, but I don't have permission to transfer.  @benjeffery?

**Reply by jeromekelleher** [reply]
Done

**Comment by jeromekelleher** [comment]
Thanks for reporting this @chlangley. I haven't looked (sorry, short on time) but I suspect that the ``verify`` command doesn't deal with missing data. If I recall, it's just checking the input genotypes against the sample data, and it may not take missing data and imputation into account.

**Reply by chlangley** [reply]
OK, thanks.  So the advise is to not attempt to use tsinfer verify on a SampleData with missing items (-1)... and forge ahead?

**Reply by jeromekelleher** [reply]
Yes, that would be my immediate reaction. Sorry, I don't have time to look at it properly right now...

This is a bug - as a minimum we should raise an error in this case saying this config isn't supported.

**Reply by jeromekelleher** [reply, marked as answer]
I've just verified that this is a bug @chlangley - I've opened  #625 to track. Thanks for reporting this!
